rotkehlchen/exchanges/bitstamp.py

```diff
@@ -689,23 +722,25 @@ def _deserialize_asset_movement_from_user_transaction(
                 f'Unexpected asset amount combination found in: {raw_movement}.',
             )
 
-        return AssetMovement(
-            timestamp=timestamp,
+        return create_asset_movement_with_fee(
             location=Location.BITSTAMP,
-            category=category,
-            address=None,  # requires query "crypto-transactions" endpoint
-            transaction_id=None,  # requires query "crypto-transactions" endpoint
+            event_type=event_type,
+            timestamp=timestamp,
             asset=fee_asset,
             amount=abs(amount),
             fee_asset=fee_asset,
-            fee=deserialize_fee(raw_movement['fee']),
-            link=str(raw_movement['id']),
+            fee=(fee := deserialize_fee(raw_movement['fee'])),
+            unique_id=(movement_id := str(raw_movement['id'])),
+            extra_data={
+                'movement_id': movement_id,
+                'fee': fee,  # Used for matching the corresponding crypto_transaction. Removed before being saved to the DB.  # noqa: E501
```

Comment:
This is super hacky. I would leave a note here and in the project notes as a TODO to rethink and fix this. This entire part of the code is really hacky

Reply:
Yes, this isn't a very elegant solution. Added a todo comment in the code and created a note: https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=90720824